File: clientel/src/app/_models/Comment.ts
export interface Comment {
    id: number;
    body: string;
    dateCreated: Date;
    dateUpdated: Date | null
    appUserId: number;
    username: string;
    photoUrl: string;
    discussionId: number;
    commentId: number | null;
    replies: Comment[]
}

File: API/Data/Entities/Comment.cs
namespace API.Data.Entities
{
    public class Comment
    {
        public int Id { get; set; }
        public string Body { get; set; }

        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }

        //Navigation
        public int AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public int DiscussionId { get; set; }
        public Discussion Discussion { get; set; }
        public int? CommentId { get; set; }
    }
}
